cylc/uiserver/app.py

```diff
@@ -515,6 +524,22 @@ def set_auth(self):
     def initialize_templates(self):
         """Change the jinja templating environment."""
 
+    def write_api_info(self):
```

Comment:
I think we can get away without this write. 
Apologies if this is stuff you already know, not sure how familiar you are with https://github.com/cylc/cylc-uiserver/pull/370

In `~/.cylc/uiserver/info_files` for every instance there are two files created.. a `jpserver-<pid>-open.html` file, which we use for opening the gui with an existing instance (see the code in `scripts/gui.py`) and another one, `jpserver-<pid>.json` it is this one that contains the same info as the api_info file.
I've compared the files and the info is the same... until you open a new instance using `cylc gui --new`, sorry I think that pr has complicated matters a little. At this point, we have one `api_info.json` file for two of the jpserver.json files. The api_info file, for me, has recorded a different port. I don't think this is a major problem but I think we can select one of these existing jpserver files and open that in the async request? This should get around the duplication.

The only potential problem I can think of at the moment is in the selection of the file. At the minute, we select a random file for re-using guis. I suspect we may want to select the particular file for the instance that we are interacting with, although I am not 100% sure about this. If so, the file name will be of the format `jpserver-<pid>.json` so we only need the process id to get the correct file.

Reply:
Sorry, I think it is a bit more complicated.

The above scenario was for running the `cylc gui` command. For `cylc hub` I didn't change the `JUPYTER_RUNTIME_DIR` variable, so the files are still generated per instance but they are still saved to the default `.local/share/jupyter` directory... see: https://docs.jupyter.org/en/latest/use/jupyter-directories.html#envvar-JUPYTER_RUNTIME_DIR

The info contained looks the same. Not sure what goes on with different users attaching though. Might need some investigation.